pool/postgres.go

```diff
@@ -234,28 +270,51 @@ func (db *PostgresDB) fetchCSRFSecret() ([]byte, error) {
 
 // persistCSRFSecret stores the bytes used for the CSRF secret in the database.
 func (db *PostgresDB) persistCSRFSecret(secret []byte) error {
+	const funcName = "persistCSRFSecret"
 	_, err := db.DB.Exec(insertCSRFSecret, hex.EncodeToString(secret))
+	if err != nil {
+		desc := fmt.Sprintf("%s: unable to persist CSRF secrets: %v",
+			funcName, err)
+		return errs.DBError(errs.PersistEntry, desc)
+	}
 	return err
 }
 
 // persistLastPaymentInfo stores the last payment height and paidOn timestamp
 // in the database.
 func (db *PostgresDB) persistLastPaymentInfo(height uint32, paidOn int64) error {
+	const funcName = "persistLastPaymentInfo"
 	tx, err := db.DB.Begin()
 	if err != nil {
 		return err
 	}
 
 	_, err = tx.Exec(insertLastPaymentHeight, height)
 	if err != nil {
-		tx.Rollback()
-		return err
+		rErr := tx.Rollback()
```

Comment:
Nice catch finding this, I didnt notice that `Rollback` could return an error.

I'm not sure about the way it is handled though. If we have a `tx.Exec` error and a `tx.Rollback` error, only the `Rollback` error is being returned and logged. The `Exec` error, which could be the root cause of the issue, will be lost. I think we need to log both.